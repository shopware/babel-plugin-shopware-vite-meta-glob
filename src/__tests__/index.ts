import pluginTester from 'babel-plugin-tester'
import plugin from '..'

function withFileName(code: string) {
  return {
    code,
    babelOptions: {filename: __filename},
  }
}

pluginTester({
  plugin,
  pluginName: 'vite-meta-glob',
  snapshot: true,
  tests: {
    // Positive tests
    'should transform glob without options': withFileName(
      'const modules = import.meta.glob("./fixtures/**/*")',
    ),
    'should transform glob with eager option': withFileName(
      'const modules = import.meta.glob("./fixtures/**/*", { eager: true })',
    ),
    'should transform glob with eager and import option': withFileName(
      'const modules = import.meta.glob("./fixtures/**/*", { eager: true, import: "default" })',
    ),

    // Negative tests
    'should not transform glob with first argument being an array':
      withFileName(
        'const modules = import.meta.glob(["./fixtures/**/*", "./fixtures/specific.js"])',
      ),
    'should not transform glob with eager option being false': withFileName(
      'const modules = import.meta.glob("./fixtures/**/*", { eager: false })',
    ),
    'should not transform other import.meta properties': withFileName(
      'const modules = import.meta.env("./fixtures/**/*")',
    ),
    'should not transform glob with import option without eager': withFileName(
      'const modules = import.meta.glob("./fixtures/**/*", { import: "default" })',
    ),
  },
})
