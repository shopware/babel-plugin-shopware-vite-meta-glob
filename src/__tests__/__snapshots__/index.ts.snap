// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vite-meta-glob should not transform glob with eager option being false: should not transform glob with eager option being false 1`] = `

const modules = import.meta.glob("./fixtures/**/*", { eager: false })

      ↓ ↓ ↓ ↓ ↓ ↓

const modules = import.meta.glob("./fixtures/**/*", {
  eager: false,
});


`;

exports[`vite-meta-glob should not transform glob with first argument being an array: should not transform glob with first argument being an array 1`] = `

const modules = import.meta.glob(["./fixtures/**/*", "./fixtures/specific.js"])

      ↓ ↓ ↓ ↓ ↓ ↓

const modules = import.meta.glob(["./fixtures/**/*", "./fixtures/specific.js"]);


`;

exports[`vite-meta-glob should not transform glob with import option without eager: should not transform glob with import option without eager 1`] = `

const modules = import.meta.glob("./fixtures/**/*", { import: "default" })

      ↓ ↓ ↓ ↓ ↓ ↓

const modules = import.meta.glob("./fixtures/**/*", {
  import: "default",
});


`;

exports[`vite-meta-glob should not transform other import.meta properties: should not transform other import.meta properties 1`] = `

const modules = import.meta.env("./fixtures/**/*")

      ↓ ↓ ↓ ↓ ↓ ↓

const modules = import.meta.env("./fixtures/**/*");


`;

exports[`vite-meta-glob should transform glob with eager and import option: should transform glob with eager and import option 1`] = `

const modules = import.meta.glob("./fixtures/**/*", { eager: true, import: "default" })

      ↓ ↓ ↓ ↓ ↓ ↓

const modules = (function () {
  const __glob__0_0 = require("./fixtures/file1.ts").default;
  const __glob__0_1 = require("./fixtures/file2.ts").default;
  const __glob__0_2 = require("./fixtures/file3.ts").default;
  const __glob__context__ = {
    "./fixtures/file1.ts": __glob__0_0,
    "./fixtures/file2.ts": __glob__0_1,
    "./fixtures/file3.ts": __glob__0_2,
  };
  return __glob__context__;
})();


`;

exports[`vite-meta-glob should transform glob with eager option: should transform glob with eager option 1`] = `

const modules = import.meta.glob("./fixtures/**/*", { eager: true })

      ↓ ↓ ↓ ↓ ↓ ↓

const modules = (function () {
  const __glob__0_0 = require("./fixtures/file1.ts");
  const __glob__0_1 = require("./fixtures/file2.ts");
  const __glob__0_2 = require("./fixtures/file3.ts");
  const __glob__context__ = {
    "./fixtures/file1.ts": __glob__0_0,
    "./fixtures/file2.ts": __glob__0_1,
    "./fixtures/file3.ts": __glob__0_2,
  };
  return __glob__context__;
})();


`;

exports[`vite-meta-glob should transform glob without options: should transform glob without options 1`] = `

const modules = import.meta.glob("./fixtures/**/*")

      ↓ ↓ ↓ ↓ ↓ ↓

const modules = {
  "./fixtures/file1.ts": () => import("./fixtures/file1.ts"),
  "./fixtures/file2.ts": () => import("./fixtures/file2.ts"),
  "./fixtures/file3.ts": () => import("./fixtures/file3.ts"),
};


`;
